Objects/Converters/ConverterCSI/ConverterCSIShared/Models/Element1DAnalyticalResultConverter.cs

```diff
@@ -0,0 +1,331 @@
+    {
+      int forcesSuccess = -1;
+
+      // Reference variables for CSI API
+      int numberOfResults = 0;
+      string[] storyName,
+        pierName,
+        loadCase,
+        location;
+      storyName = pierName = loadCase = location = new string[1];
+      double[] p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3;
+      p = v2 = v3 = t = m2 = m3 = new double[1];
+
+      if (SendForces(element1D.type))
+      {
+        forcesSuccess = sapModel.Results.PierForce(
+          ref numberOfResults,
+          ref storyName,
+          ref pierName,
+          ref loadCase,
+          ref location,
+          ref p,
+          ref v2,
+          ref v3,
+          ref t,
+          ref m2,
+          ref m3
+        );
+      }
+
+      return CreateLoadCombinationResults(
+        element1D,
+        forcesSuccess,
+        numberOfResults,
+        pierName,
+        loadCase,
+        Return0Position,
+        p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3);
+    }
+    private ICollection<ResultSet1D> GetAnalysisResultsForSpandrel(Element1D element1D)
+    {
+      int forcesSuccess = -1;
+
+      // Reference variables for CSI API
+      int numberOfResults = 0;
+      string[] storyName,
+        spandrelName,
+        loadCase,
+        location;
+      storyName = spandrelName = loadCase = location = new string[1];
+      double[] p,
+        v2,
+        v3,
+        t,
+        m2,
+        m3;
+      p = v2 = v3 = t = m2 = m3 = new double[1];
+
+      if (SendForces(element1D.type))
+      {
+        forcesSuccess = sapModel.Results.SpandrelForce(
+          ref numberOfResults,
+          ref storyName,
+          ref spandrelName,
+          ref loadCase,
+          ref location,
+          ref p,
+          ref v2,
+          ref v3,
+          ref t,
+          ref m2,
+          ref m3
+        );
+      }
+
+      return CreateLoadCombinationResults(
+        element1D, 
+        forcesSuccess, 
+        numberOfResults, 
+        spandrelName, 
+        loadCase,
+        Return0Position,
+        p, 
+        v2, 
+        v3, 
+        t, 
+        m2, 
+        m3);
+    }
+
+    private ICollection<ResultSet1D> CreateLoadCombinationResults(
+      Element1D element1D,
+      int forcesSuccess,
+      int numberOfResults,
+      string[] names,
+      string[] loadCase,
+      Func<int, float> positionCalculator, 
+      double[] p,
+      double[] v2,
+      double[] v3,
+      double[] t,
+      double[] m2,
+      double[] m3)
+    {
+      Dictionary<string, ResultSet1D> loadCombinationResults = new();
+      for (int i = 0; i < numberOfResults; i++)
+      {
+        if (names != null && names[i] != element1D.name)
+        {
+          continue;
+        }
+        Result1D result = new();
+
+        if (forcesSuccess.IsSuccessful())
+        {
+          result.position = positionCalculator(i);
+          result.forceX = (float)p[i];
```

Comment:
Not sure I follow the casting - unnecessary?

Also, these all seem to be aligned by i on the RHS, is this coming from the API?